### PR TITLE
feat: errors

### LINKED ISSUE #6: feat: Errors
In this example, we will introduce errors and how we can throw them during the execution of a contract, undoing all changes made during the transaction.
- The `assert` corelib function can be used to check a boolean condition and return an error message if it fails.
- The `panic` corelib function can be used to stop the execution of a program and return an error message.

Solidity reference: https://solidity-by-example.org/error/

### PR DESCRIPTION
**Pull request by julio4**
Close #6 
I'm considering adding another practical example of a simple vault contract with a deposit and withdraw function. Please let me know if you think it's a good idea.

Additionally, I have included some security considerations that I believe are important to keep in mind. Please let me know if you find this kind of content useful.

### CODE REVIEW

**Comment by julio4** [comment]
> This is very good, however I'm a bit worried that this is a bit out of scope of just `CairoByExample` and some of this content should be in the cairo book instead.

I agree. I might want to explain too much because I'm also learning at the same time. I'll try to keep things simple and move in-depth explanations to the Cairo book if I think it's interesting content.

**Comment by enitrat** [comment]
> I'll try to keep things simple and move in-depth explanations to the Cairo book if I think it's interesting content.

🔥

File: src/introduction/errors/errors.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Errors
+
+In smart contracts, errors play an important role during the execution of transactions. When an error is thrown, it stops the execution of the contract and all changes made during the transaction are reverted.
```
Comment (enitrat) [inline src/introduction/errors/errors.md:3]: > In smart contracts, errors play an important role during the execution of transactions.

This sentence doesn't really make sense

**Review by enitrat — APPROVED** [review]
LGTM 🔥

**Pull request merged**